pioinstaller/penv.py

```
@@ -16,18 +16,151 @@
+
+    # Handle different arm64 representations on macOS
+    if system == "Darwin" and machine in ("arm64", "aarch64"):
+        machine = "arm64"
+
+    key = (system, machine)
+    plat = platform_map.get(key)
+    # Detect musl on Linux
+    if system == "Linux" and plat and "-unknown-linux-gnu" in plat:
+        try:
+            out = subprocess.check_output(
+                ["ldd", "--version"], stderr=subprocess.STDOUT
```

Ruff (0.12.2)

Starting a process with a partial executable path

(S607)